### PR TITLE
School library: set up associations

### PR DESCRIPTION
**Pull request by theodoku**
In this Pull Request:

**- [ ] Create a class Classroom with the following:**
@label instance variable, Setter and getter for @label 
Create the has-many/belongs-to relationship between Classroom and Student

**- [ ] Create a class Book with the following:**
@title and @author instance variables, Setters and getters for instance variables 

**- [ ] Create a class Rental with the following:**
@date instance variable, Setter and getter for @date 
Create the many-to-many relationship between Person and Book using the intermediate class Rental

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -2,24 +2,29 @@
 
 class Person < Nameable
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (omar-muhamad) [inline person.rb]: - [ ] Please add Getters and Setters for @rentals, not getters only as you did. So the add rental method in the bottom can access and add a new rental.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @theodoku,

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please add a method to the student class to address the relationship when setting the classroom for a student it also adds it to the classrooms' students.
[Hint] It will be like that:

```ruby
attr_reader :classroom
  .
  .
  def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```


- [x] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: person.rb (hunk 1 above)
Comment (theodoku) [reply person.rb]: suggested changes implemented👍

**Review by rustam-tolipov — APPROVED** [review]
Hi @theodoku, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Developer-512px-10

## Highlights
1. No Linters errors 💯
2. All requirements are complete 💯
3. Well-implemented code 🔢 
4. Great approach 🥇 

## Optional suggestions
No optional suggestions :)

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**